rust-src/wallet_library/src/credential.rs

```diff
@@ -111,6 +117,102 @@ fn build_policy(
     })
 }
 
+/// The details for a new credential deployment.
+#[derive(SerdeSerialize, SerdeDeserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CredentialDeploymentDetails {
+    expiry:       TransactionTime,
+    unsigned_cdi: UnsignedCredentialDeploymentInfo<IpPairing, ArCurve, AttributeKind>,
+}
+
+/// Defines the byte serialization of the credential deployment details. Note
+/// that this serialization is for a new credential being deployed, which the
+/// explicit 0 byte denotes.
+impl Serial for CredentialDeploymentDetails {
+    fn serial<B: Buffer>(&self, out: &mut B) {
+        self.unsigned_cdi.serial(out);
+        0u8.serial(out);
+        self.expiry.seconds.serial(out);
+    }
+}
+
+/// Serializes the credential deployment details. It is this serialization that
+/// should be signed when creating a new credential for deployment. The
+/// serialized bytes are returned hex encoded.
+pub fn serialize_credential_deployment_details(
+    credential_deployment_details: CredentialDeploymentDetails,
+) -> String {
+    let mut serialized_details = Vec::<u8>::new();
+    credential_deployment_details.serial(&mut serialized_details);
+    hex::encode(serialized_details)
+}
```

Comment:
Do we need the actual data, or do we just need the hash?

There is this helper function
```rust
|pub fn credential_hash_to_sign<                                                                                                                                
            }                                                                                                                                                 |    P: Pairing,                                                                                                                                                
            return Ok(());                                                                                                                                    |    C: Curve<Scalar = P::ScalarField>,                                                                                                                         
        }                                                                                                                                                     |    AttributeType: Attribute<C::Scalar>,                                                                                                                       
    }                                                                                                                                                         |>(                                                                                                                                                             
                                                                                                                                                              |    values: &CredentialDeploymentValues<C, AttributeType>,                                                                                                     
    if let Some(wallet_path) = wallet_path {                                                                                                                  |    proofs: &IdOwnershipProofs<P, C>,                                                                                                                          
        let wallet = WalletAccount::from_json_file(wallet_path)?;                                                                                             |    new_or_existing: &Either<TransactionTime, AccountAddress>,                                                                                                 
        let param = OwnedParameter::from_serial(data: &weights)?;                                                                                             |) -> Vec<u8>
```

which would be ideal to use instead of redefining.

Reply:
We just need the hash. I am not sure why I didn't just have it return that. I tried looking through concordium-base for an existing function like `credential_hash_to_sign` but failed to find it, so it is great that you pointed out that one already exists. I will refactor to use that.